web/ui/react-app/src/QueryStats.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { PureComponent } from 'react';
+import './QueryStats.css';
+
+interface QueryStatsProps {
+    stats: {
+        loadTime: number;
+        resolution: number;
+        totalTimeSeries: number;
+    } | null;
+    loading: boolean;
+}
+
+class QueryStats extends PureComponent<QueryStatsProps>{
+
+    render() {
+        const stats = this.props.stats;
```

Comment:
Just a personal preference. I would destruct the values
const { stats, loading } = this.props
const { loadTime, resolution, totalTimeSeries } = stats